cornac/models/hrdr/hrdr.py

```diff
@@ -80,118 +179,61 @@ def __init__(self, n_users, n_items, vocab, global_mean,
                 print("Number of OOV words: %d" % oov_count)
 
         embedding_matrix = initializers.Constant(embedding_matrix)
-        i_user_id = Input(shape=(1,), dtype="int32", name="input_user_id")
-        i_item_id = Input(shape=(1,), dtype="int32", name="input_item_id")
-        i_user_rating = Input(shape=(self.n_items), dtype="float32", name="input_user_rating")
-        i_item_rating = Input(shape=(self.n_users), dtype="float32", name="input_item_rating")
-        i_user_review = Input(shape=(None, self.max_text_length), dtype="int32", name="input_user_review")
-        i_item_review = Input(shape=(None, self.max_text_length), dtype="int32", name="input_item_review")
-        i_user_num_reviews = Input(shape=(1,), dtype="int32", name="input_user_number_of_review")
-        i_item_num_reviews = Input(shape=(1,), dtype="int32", name="input_item_number_of_review")
-
-        l_user_review_embedding = layers.Embedding(self.n_vocab, self.embedding_size, embeddings_initializer=embedding_matrix, mask_zero=True, name="layer_user_review_embedding")
-        l_item_review_embedding = layers.Embedding(self.n_vocab, self.embedding_size, embeddings_initializer=embedding_matrix, mask_zero=True, name="layer_item_review_embedding")
-        l_user_embedding = layers.Embedding(self.n_users, self.id_embedding_size, embeddings_initializer="uniform", name="user_embedding")
-        l_item_embedding = layers.Embedding(self.n_items, self.id_embedding_size, embeddings_initializer="uniform", name="item_embedding")
-
-        user_bias = layers.Embedding(self.n_users, 1, embeddings_initializer=tf.initializers.Constant(0.1), name="user_bias")
-        item_bias = layers.Embedding(self.n_items, 1, embeddings_initializer=tf.initializers.Constant(0.1), name="item_bias")
-
-        user_text_processor = TextProcessor(self.max_text_length, filters=self.n_filters, kernel_sizes=self.kernel_sizes, dropout_rate=self.dropout_rate, name='user_text_processor')
-        item_text_processor = TextProcessor(self.max_text_length, filters=self.n_filters, kernel_sizes=self.kernel_sizes, dropout_rate=self.dropout_rate, name='item_text_processor')
-
-        user_review_h = user_text_processor(l_user_review_embedding(i_user_review))
-        item_review_h = item_text_processor(l_item_review_embedding(i_item_review))
+        self.graph = Model(
```

Comment:
This is more like a `build_graph` function inside HDRModel class than initializing a new instance. We only build this graph once for each HDR model, don't we?